src/graphql/fetchAllRequests.ts

```diff
@@ -0,0 +1,89 @@
+import { GraphQLClient } from 'graphql-request';
+import colors from 'colors';
+import { REQUESTS } from './gqls';
+import { makeGraphQLRequest } from './makeGraphQLRequest';
+import {
+  RequestAction,
+  RequestStatus,
+  IsoCountryCode,
+  IsoCountrySubdivisionCode,
+} from '@transcend-io/privacy-types';
+import { logger } from '../logger';
+import { LanguageKey } from '@transcend-io/internationalization';
+
+export interface PrivacyRequest {
+  /** ID of request */
+  id: string;
+  /** Time request was made */
+  createdAt: string;
+  /** Email of request */
+  email: string;
+  /** Link for request */
+  link: string;
+  /** Whether request is in test mode */
+  isTest: boolean;
+  /** Request details */
+  details: string;
+  /** Locale of request */
+  locale: LanguageKey;
+  /** Whether request is in silent mode */
+  isSilent: boolean;
+  /** Core identifier of request */
+  coreIdentifier: string;
+  /** Type of request action */
+  type: RequestAction;
+  /** Type of data subject */
+  subjectType: string;
+  /** Country of request */
+  country?: IsoCountryCode | null;
+  /** Sub division of request */
+  countrySubDivision?: IsoCountrySubdivisionCode | null;
+}
+
+const PAGE_SIZE = 50;
+
+/**
+ * Fetch all requests matching a set of filters
+ *
+ * @param client - GraphQL client
+ * @param options - Filter options
+ * @returns List of requests
+ */
+export async function fetchAllRequests(
+  client: GraphQLClient,
+  {
+    actions,
+    statuses,
+  }: {
+    /** Actions to filter on */
+    actions: RequestAction[];
+    /** Statuses to filter on */
+    statuses: RequestStatus[];
+  },
+): Promise<PrivacyRequest[]> {
+  const requests: PrivacyRequest[] = [];
+  let offset = 0;
+
+  // Paginate
+  let shouldContinue = false;
+  do {
+    const {
+      requests: { nodes, totalCount },
+      // eslint-disable-next-line no-await-in-loop
+    } = await makeGraphQLRequest(client, REQUESTS, {
+      first: PAGE_SIZE,
+      offset,
```

Comment:
We should be using cursor based pagination here using something like `createdAt` to avoid stressing out our DB on deep pages, similar to the Salesloft issue we caused, all those moons ago. An example we have in our codebase: https://github.com/transcend-io/main/blob/ceacc7b526a62c89f524ffb58ac7ae4f421d1f59/frontend-services/admin-dashboard/src/PrivacyRequests/ExportToCsv/index.tsx#L134

Reply:
oh good callout. this repo has this pagination in a future place, ill make a ticket to refactor these all, but im gonna merge this for now to unblock testing. refactoring this will require new testing.